pkg/scheduler/framework/frameworkutils.go

```diff
@@ -341,3 +341,16 @@ func sortByClusterScoreAndName(bindings []*fleetv1beta1.ClusterResourceBinding)
 
 	return bindings
 }
+
+// shouldSchedule checks if the scheduler needs to perform some scheduling.
+//
+// A scheduling cycle is only needed if
+// * the policy is of the PickAll type; or
+// * the policy is of the PickN type, and currently there are not enough number of bindings.
+func shouldSchedule(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot, desiredCount, existingCount int) bool {
+	if policy.Spec.Policy.PlacementType == fleetv1beta1.PickAllPlacementType {
```

Comment:
check if policy.Spec.Policy is nil?

Reply:
Ah, this method is only called in the PicKN process branch, which is guaranteed to have a policy. But for the clarity I'll add a check.